src/checkpoint/dispatch/dispatch_serializer_nonbyte.h

```diff
@@ -161,6 +170,23 @@ struct SerializerDispatchNonByte {
     serializeEnum(s, *val);
   }
 
+  template <typename U = T>
+  void applyStatic(
+    SerializerT& s, T* val, SerialSizeType num, justFootprint<U>* = nullptr
+  ) {
+    debug_checkpoint(
+      "SerializerDispatch: justFootprint: val=%p\n",
+      static_cast<void*>(&val)
+    );
+    static bool firstCall = true;
+    if (firstCall) {
+      firstCall = false;
+      // optionally print a warning here
```

Comment:
We need to decide what to do here.

Reply:
hard to say how much additional value a separate warning would provide, maybe expanding the first (unconditional) print on the entrance would be sufficient (e.g. adding `typeid`)?

Reply:
My thinking is that we may want a printed warning here in much broader circumstances than when we have full debug printing turned on. Perhaps aligned with `CMAKE_BUILD_TYPE=Debug` or such.

Reply:
Regardless, both this debug print and the ones below should probably have a name from `typeid` incorporated.

Reply:
I agree that printing `typeid` would be good here.